### PR TITLE
Added PROCMON plugin for watching life cycle of the processes.

### PR DESCRIPTION
**Pull request by disaykin**
Currently supports only process creation via NtCreateUserProcess
and only on x64 arch.

### CODE REVIEW

File: src/plugins/procmon/procmon.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#ifndef PROCMON_H
+#define PROCMON_H
+
+#include "plugins/private.h"
+#include "plugins/plugins.h"
+
+class procmon: public plugin
+{
+public:
+    drakvuf_trap_t traps[1] =
```
Comment (tklengyel) [inline src/plugins/procmon/procmon.h]: Don't have to make an array for just a single trap.. or do you expect more traps to be added here in the future?

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test.

File: src/plugins/procmon/procmon.h (hunk 1 above)
Comment (disaykin) [reply src/plugins/procmon/procmon.h]: Yes. Process termination, for example.

**Comment by disaykin** [comment]
Rebase + add support of Time field from last commit from @arkfps

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please.

**Comment by tklengyel** [comment]
Looks good!

**Pull request merged**